code/modules/mapping/map_template.dm

```diff
@@ -26,8 +26,11 @@
 	var/list/obj/structure/cable/cables = list()
 	var/list/atom/atoms = list()
 
-	for(var/L in block(locate(bounds[MAP_MINX], bounds[MAP_MINY], bounds[MAP_MINZ]),
-	                   locate(bounds[MAP_MAXX], bounds[MAP_MAXY], bounds[MAP_MAXZ])))
+	var/list/turfs = block(	locate(bounds[MAP_MINX], bounds[MAP_MINY], bounds[MAP_MINZ]),
+							locate(bounds[MAP_MAXX], bounds[MAP_MAXY], bounds[MAP_MAXZ]))
+	var/list/border = block(locate(bounds[MAP_MINX]-1, bounds[MAP_MINY]-1, bounds[MAP_MINZ]),
+							locate(bounds[MAP_MAXX]+1, bounds[MAP_MAXY]+1, bounds[MAP_MAXZ])) - turfs
```

Comment:
Won't this have issues with templates loaded on the edges of maps?

Reply:
it will, I'll fix